backend/app/api/endpoints/knowledge.py

```
@@ -775,6 +782,128 @@ async def create_document(
+            db=db,
+            user=current_user,
+            knowledge_base_id=knowledge_base_id,
+            provider_id=data.provider,
+            external_resource_id=data.external_resource_id,
+            folder_id=data.folder_id,
+        )
+    except ExternalDocumentImportError as exc:
+        raise HTTPException(
+            status_code=exc.status_code,
+            detail=str(exc),
+        )
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)